### PR TITLE
MAT-8414, 15, 16 Add a minimize button and it's actions to MadieAlert

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR

Jira Ticket: [MAT-8414](https://jira.cms.gov/browse/MAT-8414), [MAT-8415](https://jira.cms.gov/browse/MAT-8415), [MAT-8416](https://jira.cms.gov/browse/MAT-8416)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: react/test/components/MadieAlert.test.js [hunk 1]
```diff
@@ -271,4 +272,294 @@ Row: 6, Col:0: VSAC: 0:87 | Request failed with status code 404 for oid = 1.16.8
...
+        fireEvent.click(minimizeButton2);
+
+        // Now both alerts are minimized, check for a count of 3
+        await waitFor(() => {
+            expect(getByTestId("minimized-alert-text")).toHaveTextContent("Display Alerts (3)");
+        });
+    });
+
+    it("copies content from all alerts with copyButton enabled", async () => {
+        const expectedText = `Error 1\nError 2\n\nWarning 1`;
```
Comment (etan-sb) [inline react/test/components/MadieAlert.test.js:440]: Is `expectedText` being used? Should this test be checking to make sure we copy all text content from all alerts?

File: react/test/components/MadieAlert.test.js [hunk 2]
```diff
@@ -271,4 +272,294 @@ Row: 6, Col:0: VSAC: 0:87 | Request failed with status code 404 for oid = 1.16.8
...
+
+    it("maintains backward compatibility with original API", () => {
+        const { getByTestId } = render(
+            <MadieAlert
+                type="error"
+                visible={true}
+                canClose={false}
+                content={<h1>Legacy API</h1>}
+                alertProps={{
+                    "data-testid": "legacy-alert",
```
Comment (etan-sb) [inline react/test/components/MadieAlert.test.js:499]: I can't seem to find "legacy-alert" anywhere. Do you have an example of calling MadieAlert with this data-testid in our repos?

File: react/test/components/MadieAlert.test.js [hunk 3]
```diff
@@ -271,4 +272,294 @@ Row: 6, Col:0: VSAC: 0:87 | Request failed with status code 404 for oid = 1.16.8
...
+        );
+
+        // Update to use the index-specific button ID
+        const minimizeButton = getByTestId("minimize-button-0");
+        fireEvent.click(minimizeButton);
+
+        // Check that there's no error count
+        await waitFor(() => {
+            expect(getByTestId("minimized-alert-text")).toHaveTextContent("Display Alerts");
+            expect(getByTestId("minimized-alert-text")).not.toHaveTextContent("Display Alerts (");
```
Comment (etan-sb) [inline react/test/components/MadieAlert.test.js:562]: Should this just be `expect(getByTestId("minimized-alert-text")).not.toHaveTextContent("Display Alerts");` (with no '(' character at the end)?

Comment (ethankaplan) [reply react/test/components/MadieAlert.test.js:562]: It conditionally renders the parentheses, so it's making sure it's not rendering them here
